### PR TITLE
CASMINST-4924: move argo templates into docs

### PR DESCRIPTION
**Pull request by dzou-hpe**
# Description

move argo templates into docs

https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-4924

# Checklist Before Merging

- [ ] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I don't have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Review by alexanderkingh — APPROVED** [review]

**Comment by heemstra** [comment]
It'd be better to have the bash in individual script files so they can be linted - and then have the yaml invoke the scripts.

**Comment by dzou-hpe** [comment]
> It'd be better to have the bash in individual script files so they can be linted - and then have the yaml invoke the scripts.

Agreed, the middle term plan is to reduce bash to minimum and long term plan is to get rid of bash here. We are also trying to avoid ssh as much as possible so we don't want to put it on disk. I will need to develop a new way for storing bash outside then `import` it on the fly.

File: workflows/ncn/worker/common.envar.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+TARGET_NCN={{ `{{inputs.parameters.targetNcn}}` }}
+TARGET_XNAME=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/apis/sls/v1/search/hardware?extra_properties.Role=Management" | \
+     jq -r ".[] | select(.ExtraProperties.Aliases[] | contains(\"$TARGET_NCN\")) | .Xname")
+TARGET_MGMT_XNAME=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/apis/sls/v1/search/hardware?extra_properties.Role=Management" | \
+  jq -r ".[] | select(.ExtraProperties.Aliases[] | contains(\"$TARGET_NCN\")) | .Parent")
+
+TARGET_NCN_mgmt_host="${TARGET_NCN}-mgmt"
+
+
+export IPMI_USERNAME=$(curl -XGET -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/apis/scsd/v1/bmc/creds"| jq -r ".Targets[] | select(.Xname | contains(\"$TARGET_MGMT_XNAME\")) | .Username")
```
Comment (mharding-hpe) [inline workflows/ncn/worker/common.envar.yaml:41]: I don't know if the export in this argo template is the same as a normal Bash export, but if so, I don't think we need to export the username. Only the password needs to be exported.

Comment (mharding-hpe) [inline workflows/ncn/worker/common.envar.yaml:41]: I don't know if the export in this argo template is the same as a normal Bash export, but if so, I don't think we need to export the username. Only the password needs to be exported.

```suggestion
IPMI_USERNAME=$(curl -XGET -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/apis/scsd/v1/bmc/creds"| jq -r ".Targets[] | select(.Xname | contains(\"$TARGET_MGMT_XNAME\")) | .Username")
```

**Comment by mharding-hpe** [comment]
> > It'd be better to have the bash in individual script files so they can be linted - and then have the yaml invoke the scripts.
> 
> Agreed, the middle term plan is to reduce bash to minimum and long term plan is to get rid of bash here. We are also trying to avoid ssh as much as possible so we don't want to put it on disk. I will need to develop a new way for storing bash outside then `import` it on the fly.

But as long as we still have Bash, would it not be possible to do as John suggests?

**Comment by dzou-hpe** [comment]
> > > It'd be better to have the bash in individual script files so they can be linted - and then have the yaml invoke the scripts.
> > 
> > 
> > Agreed, the middle term plan is to reduce bash to minimum and long term plan is to get rid of bash here. We are also trying to avoid ssh as much as possible so we don't want to put it on disk. I will need to develop a new way for storing bash outside then `import` it on the fly.
> 
> But as long as we still have Bash, would it not be possible to do as John suggests?

not at the moment and we have ticket open for revisit all the bash we use here but we never have time to do it. I'd suggest bringing this up in standup, if we can have more time on this, i'd love to work on that.

File: workflows/ncn/worker/worker.rebuild.yaml [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+              parameters:
+              - name: targetNcn
+                value: {{$value}}
+              - name: dryRun
+                value: "{{$.DryRun}}"
+          # wipe and reboot: parallel
+          #     once a worker node is drained from k8s
+          #     we can safely wipe and reboot this node
+          #     regardless of what state other nodes are
+          - name: wipe-and-reboot-{{$value}}
```
Comment (rustydb) [inline workflows/ncn/worker/worker.rebuild.yaml:188]: We don't have to wipe anymore, we can though.

This should probably just be "set nextboot"

It should just set pxe for the next boot, and maybe disable the disk bootloader.

For reference the rebuild script in main is updated to do just this, I'd link it for reference right now if I could but I'm mobile. I'll link it Monday unless someone beats me to it.

**Review by rustydb — CHANGES_REQUESTED** [review]

File: workflows/ncn/worker/worker.rebuild.yaml (hunk 2 above)
Comment (rustydb) [reply workflows/ncn/worker/worker.rebuild.yaml:188]: Wiping would ensure a pxe boot, but the reason I care is I want to remove the mentality that a wipe is required. The deployment code will wipe. If a wipe fails then it's the fault of the deployment code, and the deployment code handles wiping better than the manual/argo step(s).

Comment (dzou-hpe) [reply workflows/ncn/worker/worker.rebuild.yaml:188]: i still have that ticket assigned to me. the reason why i kept this for now is so i can test on a 1.2 system until we get all 1.3 upgrade script cleaned up

**Review by rustydb — APPROVED** [review]

**Pull request merged**